samples/azure/azure-service-bus-long-running/ASB_7/Endpoint2/LongProcessingRequestSaga.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage.Table;
+using NServiceBus;
+using NServiceBus.Logging;
+using Shared;
+
+public class LongProcessingRequestSaga : Saga<SagaState>, 
+                                            IAmStartedByMessages<LongProcessingRequest>, 
+                                            IHandleTimeouts<ProcessingPossiblyFailed>,
+                                            IHandleMessages<LongProcessingFinished>,
+                                            IHandleMessages<LongProcessingFailed>
+{
+    static ILog log = LogManager.GetLogger<LongProcessingRequestSaga>();
+    public CloudTable table;
+
+    public LongProcessingRequestSaga()
+    {
+        var cloudTableClient = StorageHelper.GetTableClient();
+        table = cloudTableClient.GetTableReference(Constants.TableName);
+        table.CreateIfNotExists();
+    }
+
+    public async Task Handle(LongProcessingRequest message, IMessageHandlerContext context)
+    {
+        log.Info($"Received LongProcessingRequest with ID {message.Id}, EstimatedProcessingTime: {message.EstimatedProcessingTime}.");
+
+        #region setting-timeout
+
+        var timeoutToBeInvokedAt = DateTime.Now + message.EstimatedProcessingTime + TimeSpan.FromSeconds(10);
+        await RequestTimeout(context, timeoutToBeInvokedAt, new ProcessingPossiblyFailed { Id = message.Id });
+
+        #endregion
+
+        log.Info($"Timeout is set to be executed at {timeoutToBeInvokedAt}.");
+
+
+        log.Info("Registering long running request with Processor.");
+
+        #region enqueue-request-for-processor
+
+        // Saga enqueues the request in storate table. Normally, if there's more than that, work should be done by a handler
```

Comment:
How about: Saga enqueues the request to process in a storage table. This is the logical equivalant of adding a message to a queue, if there would be business specific work in here, that work should be done by sending a message to a handler instead, not in the saga.